pkg/imageverifier/bindir/bindir.go

```diff
@@ -0,0 +1,96 @@
+package bindir
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/pkg/imageverifier"
+)
+
+type Config struct {
+	Enabled bool   `toml:"enabled"`
+	BinDir  string `toml:"bin_dir"`
+}
+
+type ImageVerifier struct {
+	config *Config
+}
+
+var _ imageverifier.ImageVerifier = (*ImageVerifier)(nil)
+
+func NewImageVerifier(c *Config) *ImageVerifier {
+	return &ImageVerifier{
+		config: c,
+	}
+}
+
+func (v *ImageVerifier) Enabled(ctx context.Context) bool {
+	return v.config.Enabled
+}
+
+func (v *ImageVerifier) VerifyImage(ctx context.Context, name string, digest string) imageverifier.Judgement {
+	des, err := os.ReadDir(v.config.BinDir)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return imageverifier.Judgement{
+				OK:     true,
+				Reason: "image verification not enabled",
+			}
+		}
+
+		return imageverifier.Judgement{
+			OK:     false,
+			Reason: fmt.Sprintf("failed to list directory contents: %v", err),
+		}
+	}
+
+	if len(des) == 0 {
+		return imageverifier.Judgement{
+			OK:     true,
+			Reason: fmt.Sprintf("no image verifier binaries found in %v", v.config.BinDir),
+		}
+	}
+
+	reason := &strings.Builder{}
+	for i, de := range des {
+		bin := path.Join(v.config.BinDir, de.Name())
+		args := []string{
+			"-name", name,
+			"-digest", digest,
+		}
+
+		cmd := exec.CommandContext(ctx, bin, args...)
```

Comment:
We should probably set a timeout on this command execution.  (That might be something worth making configurable too.)